safekeeper/src/timeline.rs

```diff
@@ -101,7 +103,14 @@ impl SharedState {
     ) -> Result<Self> {
         let state = SafeKeeperState::new(zttid, peer_ids);
         let control_store = control_file::FileStorage::create_new(zttid, conf, state)?;
-        let wal_store = wal_storage::PhysicalStorage::new(zttid, conf);
+
+        let segment_complete = watch::channel((Lsn::MAX, Lsn::MAX));
+        let lsn_persisted = watch::channel(Lsn::MAX);
+        let wal_store = wal_storage::PhysicalStorage::new(zttid, conf, lsn_persisted.0, segment_complete.0);
+
+        let remote_storage = remote_storage::create(conf.remote_storage_config.as_ref(), conf.timeline_dir(zttid))?;
```

Comment:
Current remote storage limits s3 requests using an internal semaphore, so multiple remote storages would have their own, independent limits.
This is odd and I think there should be only one `Arc<RemoteStorage>` stored somewhere on a top level (safekeper main file?) and shared among the rest of the code.
Every method in the `RemoteStorage` requires `&self` so it's fine to be used within `Arc`s.

Also, for pageserver's LocalStorage at least, we've intended its root to be the directory with the `tenants` folder.

Reply:
re. semaphore: as discussed via Slack, throttling implementation with semaphore is probably not optimal. This is exacerbated by the fact we're throttled by IAM while we access S3 (IAM control plane is in USE1, which poses an availability issue in addition to much lower API request limits as compared to S3).